frontend/src/main/web/lib/stores/GlossaryStore.js

```diff
@@ -0,0 +1,522 @@
+            resolve(res['body']);
+          }
+        }
+        resolve(null);
+      })
+      .on('progress', function(e) {
+        _state['uploadFile']['status'] = e.percent;
+      });
+  });
+}
+
+function processDelete(serverResponse) {
+  console.debug('Glossary entry deleted');
+}
+
+function processSaveOrUpdate(serverResponse) {
+  _state['newEntry'].isSaving = false;
+  _state['newEntry'].show = false;
+  _state['newEntry'].isAllowSave = false;
+  _state['newEntry'].pos = '';
+  _state['newEntry'].term = '';
+  _state['newEntry'].description = '';
+
+  console.debug('Glossary entry saved');
+}
+
+function processUpdate(serverResponse) {
+  console.debug('Glossary entry updated');
+}
+
+function initialise() {
+  //load permission here
+  _state['canAddNewEntry'] = canAddNewEntry();
+  _state['canUpdateEntry'] = canUpdateEntry();
+
+  loadLocalesStats()
+    .then(processLocalesStatistic)
+    .then(loadGlossaryByLocale)
+    .then(processGlossaryList)
+    .then(function (newState) {
+      GlossaryStore.emitChange();
+    });
+}
+
+var GlossaryStore = assign({}, EventEmitter.prototype, {
+  init: function() {
+    if (_state.locales === null) {
+      initialise();
+    }
+    return _state;
+  }.bind(this),
+
+  getEntry: function(resId) {
+    return _state['glossary'][resId];
+  },
+
+  getFocusedRow: function() {
+    return _state['focusedRow'];
+  },
+
+  getNewEntryState: function() {
+    return _state['newEntry'];
+  },
+
+  getUploadFileState: function() {
+    return _state['uploadFile'];
+  },
+
+  emitChange: function() {
+    this.emit(CHANGE_EVENT);
+  },
+
+  /**
+   * @param {function} callback
+   */
+  addChangeListener: function(callback) {
+    this.on(CHANGE_EVENT, callback);
+  },
+
+  /**
+   * @param {function} callback
+   */
+  removeChangeListener: function(callback) {
+    this.removeListener(CHANGE_EVENT, callback);
+  },
+
+  dispatchToken: Dispatcher.register(function(payload) {
+    var action = payload.action;
+    switch (action['actionType']) {
+      case GlossaryActionTypes.TRANS_LOCALE_SELECTED:
+        console.debug('select language', action.data);
+        _state['selectedTransLocale'] = action.data;
+        resetCache();
+        loadGlossaryByLocale()
+          .then(processGlossaryList)
+          .then(function (newState) {
+            GlossaryStore.emitChange();
+          });
+        break;
+      case GlossaryActionTypes.INSERT_GLOSSARY:
+        console.debug('save glossary', action.data);
+        _state['newEntry'].isSaving = true;
+        _state['newEntry'].show = true;
+        GlossaryStore.emitChange();
+
+        saveGlossary(action.data)
+          .then(processSaveOrUpdate)
+          .then(initialise);
+        break;
+      case GlossaryActionTypes.UPDATE_GLOSSARY:
+        console.debug('update glossary', _state['glossary'][action.data]);
+        updateGlossary(_state['glossary'][action.data])
+          .then(processUpdate)
+          .then(initialise);
```

Comment:
It seems like `initialise` is being used a lot after actions happen - the name suggests that it would only happen once at the beginning. It should have a different name that properly indicates what it actually does (maybe something about refreshing, updating or clearing things?).